dataset/st_dataset.py

```diff
@@ -292,3 +292,108 @@ def show_description(self):
+        train_set: List[Dict] = [],
+        test_set: List[Dict] = [],
+        validation_set: List[Dict] = [],
+        query_based: bool = False,
+        multi_doc: bool = False,
+        dialogue_based: bool = False,
+    ):
+        """Create dataset information from the huggingface Dataset class
+        :rtype: dataset object
+        :param train_set: List[Dict], list of dictionaries that contain a data instance.
+            Contains the training examples and is in the form listed below.
+                The dictionary is in the form:
+                    {"source": "source_data", "summary": "summary_data", "query":"query_data"}
+                        * source_data is either of type List[str] or str
+                        * summary_data is of type str
+                        * query_data is of type str
+                The list of dictionaries looks as follows:
+                    [dictionary_instance_1, dictionary_instance_2, ...]
+        :param validation_set: Optional[List[Dict]], similar format to train_set
+            Contains the validation examples
+        :param test_set: Optional[List[Dict]], similar format to train_set
+            Contains the test examples
+        :param query_based: bool, Is the dataset query-based?
+        :param multi_doc: bool, Does each dataset instance source contain multiple documents
+        :param dialogue_based: Is the dataset dialogue-based?
+        """
+
+        if not (train_set or test_set or validation_set):
+            raise ValueError("Missing data for the dataset")
+
+        self.is_dialogue_based = dialogue_based
+        self.is_multi_document = multi_doc
+        self.is_query_based = query_based
+
+        # Load the data into their respective splits
+        self._train_set = self._process_data(train_set, "Train")
+        self._validation_set = self._process_data(validation_set, "Validation")
+        self._test_set = self._process_data(test_set, "Test")
+
+        self.dataset_name = None
+        self.version = None
+
+        self.dataset_name = "Custom"
+
+    def _process_data(
+        self, data: Dataset, split: str
+    ) -> Generator[SummInstance, None, None]:
+        """
+        This method processes the data contained in the dataset
+            and puts each data instance into a SummInstance object
+        :param dataset: a train/validation/test dataset
+        :rtype: a generator yielding SummInstance objects
+        """
+        data_instances = []
+
+        summaries_present = False
+        for instance in data:
+            if "source" in instance and instance["source"]:
+                if self.is_dialogue_based or self.is_multi_document:
+                    source: List = instance["source"]
+                else:
+                    source: str = instance["source"]
+            else:
+                raise TypeError("Missing source for a dataset instance")
+
+            # TODO: Ensure models can handle datasets with no summaries
```

Comment:
I think for the current models, the `summarize()` function only takes the source and query(optional), so it should be good